api/changePassword.ts

```diff
@@ -0,0 +1,31 @@
+import handler from './handler';
+import withConnection from './db';
+import * as auth from './authenticate';
+import * as errors from './errors';
+import { Client } from 'pg';
+
+// Expects oldPassword and newPassword
+// Returns user and new session
+// This logs out all existing sessions; you must change your saved session id to this new id
+export default handler(async (req, userId) => {
+    errors.requireParams(req.body, ['oldPassword', 'newPassword']);
```

Comment:
I think we need to have some sort of class verification enabled here (That way, we can refer strictly to a TS interface for the `requireParams` function. Especially if we receive more complex post requests.)
However, I'm struggling to find a good suggestion, so that will have to be a stretch goal.